parlai/core/torch_ranker_agent.py

```diff
@@ -313,9 +318,29 @@ def eval_step(self, batch):
             cand_preds_generator = (cand_list[rank] for rank in ordering)
             cand_preds.append(list(islice(cand_preds_generator, max_preds)))
 
+        if (self.opt.get('repeat_blocking_heuristic', False) and
+            self.opt.get('eval_candidates') == 'fixed'):
+            cand_preds = self.block_repeats(cand_preds)
```

Comment:
hmm. i think that we should alter "preds" instead of "cand_preds". i.e., maintain the order of cands predicted by the model, but select the prediction that was not repeated. thoughts?

Reply:
hmm.. yes.. could do it that way.. not really a big difference i guess

Reply:
actually thinking about it.. in say comment battle kurt picks from the top N to show possibilities.. so it is best if the repeat is just removed from the cands list

Reply:
ah good point